docs/examples/vectorfields.py

```diff
@@ -0,0 +1,493 @@
+#
+# $$
+# k_{\text{vel}} \left(\mathbf{X}, \mathbf{X}^{\prime}\right)= \begin{cases}k^{(z)}\left(\mathbf{x}, \mathbf{x}^{\prime}\right) & \text { if } z=z^{\prime} \\ 0 & \text { if } z \neq z^{\prime}\end{cases}
+# $$
+#
+# where $k^{(z)}\left(\mathbf{x}, \mathbf{x}^{\prime}\right)$ are the user are kernels for each dimension. What this means is that there are no correlations between the $x^{(0)}$ and $x^{(1)}$ dimensions for all choices $\mathbf{X}$ and $\mathbf{X}^{\prime}$, since there are no off-diagonal elements in the Gram matrix populated by this choice.
+#
+# To implement this approach in GPJax, we define `VelocityKernel` in the following cell, following the steps outlined in the creating new kernels notebook. This modular implementation takes the choice of user kernels as its class attributes: `kernel0` and `kernel1`. We must additionally pass the argument `active_dims = [0,1]`, which is an attribute of the base class `AbstractKernel`, into the chosen kernels. This is necessary such that the subsequent likelihood optimisation does not optimise over the artificial label dimension.
+#
+
+# %%
+from jax.numpy import exp
+
+
+@dataclass
+class VelocityKernel(gpx.kernels.AbstractKernel):
+    kernel1: gpx.kernels.AbstractKernel = gpx.kernels.RBF(active_dims=[0, 1])
+    kernel0: gpx.kernels.AbstractKernel = gpx.kernels.RBF(active_dims=[0, 1])
+
+    def __call__(
+        self, X: Float[Array, "1 D"], Xp: Float[Array, "1 D"]
+    ) -> Float[Array, "1"]:
+        # standard RBF-SE kernel is x and x' are on the same output, otherwise returns 0. acheive the correct value via 'Switches'
+
+        z = X[2]
+        zp = Xp[2]
+
+        k0Switch = (z + 1) % 2 * (zp + 1) % 2
+        k1Switch = z * zp
+
+        return k0Switch * self.kernel0(X, Xp) + k1Switch * self.kernel1(X, Xp)
+
+
+# %% [markdown]
+# ### GPJax Implementation
+# Next, we define the model in GPJax. The prior is defined using $k_{\text{vel}}\left(\mathbf{X}, \mathbf{X}^\prime \right)$ and 0 mean. We choose a Gaussian likelihood.
+#
+
+# %%
+# Define the Helmholtz GP
+kernel = VelocityKernel()
+mean = gpx.mean_functions.Zero()
+Prior = gpx.Prior(mean_function=mean, kernel=kernel)
+Likelihood = gpx.Gaussian(
+    num_datapoints=D0.n, obs_noise=jnp.array([1.0], dtype=jnp.float64)
+)
+Posterior = Prior * Likelihood
+
+
+# %% [markdown]
+# With a model now defined, we can proceed to optimise the hyperparameters of our likelihood over $D_0$. This is done by minimising the marginal log likelihood using `optax`. We also plot its value at each step to visually confirm that we have found the minimum
```

Comment:
Add a short comment here to point the reader to the intro to GPs notebook for more information on optimising the GP w.r.t the MLL